### ISSUE TITLE
[Feature] Google translate in lyrics

### ISSUE THREAD

**Issue by farzana840** [25huizengek1/ViMusic#92]
A translate lyrics feature would be really good so that we can listen to songs in other languages and not be lost looking at their lyrics. Would be really helpful. Thank you

**Comment by dtomvan** [comment]
I don't see how this would work with a FOSS app. The user would have to copy-paste an API token etc. Would also be a hassle to implement efficiently and without exceeding the quota within 10 minutes...

**Comment by farzana840** [comment]
Oh, i see. But as far as I know, another fork of vimusic and innertune both have this feature and they don't require me to enter the api(both are Foss too). Hmm...maybe they use something else to translate...? Weird

**Comment by dtomvan** [comment]
I believe I have found the ViMusic fork you are referring to: [fast4x's RiMusic](https://github.com/fast4x/rimusic) uses a Google Translate library for kotlin, which is not using the official endpoint, thus probably breaking TOS. If not then I would still question the ethics of this way of translating lyrics. Exerpt from [their README](https://github.com/therealbush/translator): 

> Google Translate is one of the better translators out there, but their API is not free. This uses an unofficial API endpoint that does not require an API token.

I would then argue that even if you have acquired a generated translation of the lyrics, it wouldn't be of any use to the user; a lot of lyrical and/or artistic meaning will be lost and it wouldn't be possible to synchronise or use along with the song in realtime in any way.

To summarise: it isn't worth it...

To any listener trying to understand the meaning of a song in a foreign language I would recommend referring to a human translation or interpretation, not something Google translate would come up with.

Thank you for your suggestion, though.

**Comment by 25huizengek1** [comment]
I appreciate this suggestion a lot, but I do want to agree with @dtomvan, this doesn't add a lot of value, I think, it only complicates things and clutters the UI, while using a hacky way of getting the job done.

If there are any ideas / thoughts on this I'd happily reconsider, but I'm closing this issue for now.
